buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -376,16 +368,16 @@ private static void configureCleanupHooks(Project project) {
         });
 
         project.getGradle().buildFinished(buildResult -> {
-            logger.info("Build finished");
+            logger.info("Build finished: {}", project.getPath());
             shutdownExecutorService();
         });
         // When the Daemon is not used, or runs into issues, rely on a shutdown hook
         // When the daemon is used, but does not work correctly and eventually dies off (e.x. due to non interruptible
         // thread in the build) process will be stopped eventually when the daemon dies.
-        Runtime.getRuntime().addShutdownHook(new Thread(TestClustersPlugin::shutDownAllClusters));
+        Runtime.getRuntime().addShutdownHook(new Thread(this::shutDownAllClusters));
```

Comment:
I think this is going to cause memory leak issues in the daemon as we now have a global lister which holds a reference to this plugin and everything with it. We should either remove this an accept that some builds that terminate awkwardly might leave dangling processes or ensure we explicitly remove this listener.

Reply:
Excellent find! I made changes to remove the listener when the build completes.